cmd/admin-server-info.go

```diff
@@ -145,3 +146,34 @@ func getLocalCPUInfo(endpoints EndpointList, r *http.Request) madmin.ServerCPUHa
 		CPUInfo: cpuHardwares,
 	}
 }
+
+// getLocalNetworkInfo - returns ServerNetworkHardwareInfo only for the
+// local endpoints from given list of endpoints
+func getLocalNetworkInfo(endpoints EndpointList, r *http.Request) madmin.ServerNetworkHardwareInfo {
+	var networkHardwares []net.Interface
+	seenHosts := set.NewStringSet()
+	for _, endpoint := range endpoints {
+		if seenHosts.Contains(endpoint.Host) {
+			continue
+		}
+		// Only proceed for local endpoints
+		if endpoint.IsLocal {
+			networkHardware, err := net.Interfaces()
+			if err != nil {
+				return madmin.ServerNetworkHardwareInfo{
+					Error: err.Error(),
+				}
+			}
+			networkHardwares = append(networkHardwares, networkHardware...)
```

Comment:
Since all the local endpoints have same network interface info. We could break the loop on the first catch.

Reply:
good catch. 
Added the below code 
```
// Add to the list of visited hosts
seenHosts.Add(endpoint.Host)
```